analyzers/src/SonarAnalyzer.CSharp/Rules/Hotspots/ClearTextProtocolsAreSensitive.cs

```diff
@@ -124,7 +124,7 @@ private static void VisitAssignments(SyntaxNodeAnalysisContext context)
         private void VisitStringExpressions(SyntaxNodeAnalysisContext c)
         {
             var text = GetText(c.Node);
-            if (GetUnsafeProtocol(text) is {} unsafeProtocol)
+            if (GetUnsafeProtocol(text, c.Node) is {} unsafeProtocol)
```

Comment:
now that you are passing `c.Node` to `GetUnsafeProtocol`, wouldn't it be better to call `GetUnsafeProtocol(c.Node)` and retrieve the text inside the method?